[SYCL] Add support for the generic address space in tests for sycl::a…

psamolysov-intel opened the pull request:
…tomic_ref

Signed-off-by: Pavel Samolysov <pavel.samolysov@intel.com>

Commit:
[SYCL] Add support for the generic address space in tests for sycl::atomic_ref

Signed-off-by: Pavel Samolysov <pavel.samolysov@intel.com>

ghost commented:
/verify with intel/llvm#5148

Commit:
[SYCL][NFC] Fix formatting issues

Signed-off-by: Pavel Samolysov <pavel.samolysov@intel.com>

Commit:
[SYCL] Add support for the generic address space in more tests

Signed-off-by: Pavel Samolysov <pavel.samolysov@intel.com>

Commit:
[SYCL][NFC] Fix formatting issues

Signed-off-by: Pavel Samolysov <pavel.samolysov@intel.com>

ghost commented:
/verify with intel/llvm#5148

ghost commented:
/verify with intel/llvm#5148

ghost commented:
/verify with intel/llvm#5148

ghost commented:
/verify with intel/llvm#5148

ghost commented:
/verify with intel/llvm#5148

ghost commented:
Very strange, during the previous run, CUDA job was not able to get the changes from git.

Commit:
[SYCL] Move tests for the generic address space in separated .cpp files

Unfortunately, the CUDA backend in DPCPP has had no generic address space
support yet. The tests for the generic address space must fail on the CUDA
backend. To let us mark these tests as expectedly failed, the tests should
be moved to separated .cpp files.

Signed-off-by: Pavel Samolysov <pavel.samolysov@intel.com>

Commit:
[SYCL] Fix compilation errors in the add and sub tests

Signed-off-by: Pavel Samolysov <pavel.samolysov@intel.com>

ghost commented:
/verify with intel/llvm#5148

ghost commented:
Some tests sporadically failed. Re-run the tests against the feature commit in sycl.

ghost commented:
/verify with intel/llvm#5148

ghost commented:
/verify with intel/llvm#5148

ghost commented:
A test failed against the https://github.com/intel/llvm/pull/5148 PR but this test is not related to the introduced changes. The PR can be merged after https://github.com/intel/llvm/pull/5148

bader commented:
@AGindinson, ping.

AGindinson commented on SYCL/AtomicRef/add.h:
```diff
@@ -159,12 +163,34 @@ void add_post_inc_test(queue q, size_t N) {
 
 template <typename T, typename Difference = T>
 void add_test(queue q, size_t N) {
```
Minor nit that could be applicable for other tests as well - but feel free to decide on whether this cosmetic change is necessary.
```suggestion
void add_global_test(queue q, size_t N) {
```

psamolysov-intel replied:
I believe it is a temporary solution just to let the tests pass with CUDA backend since the backend has had no support for the generic address space yet. When the support for generic address space is added, all the `*_generic.cpp` files will be removed and so, the name `*_test` without any markers (global, generic, etc.) will the only for use.

Commit:
[SYCL] Trigger precommit tests

ghost commented:
I'm waiting for the ongoing intel/llvm nightly-build update to trigger the checks against the compiler with the implemented feature.

bader commented:
> I'm waiting for the ongoing intel/llvm nightly-build update to trigger the checks against the compiler with the implemented feature.

I think Jenkins/llvm-test-suite checked against the compile with the implemented feature.
I see that `SYCL :: DiscardEvents/invalid_event.cpp` failed there. I'm not sure whether it's related to your patch, but if it's not, please, file a bug report and assign to @alexanderfle (he added this test [recently](https://github.com/intel/llvm-test-suite/commit/40d618354b7e11231644a14016869e9b1277222a)).

ghost commented:
@bader Thank you for the information, I've trigger the checks. I believe the issue with `SYCL :: DiscardEvents/invalid_event.cpp` is not related to the feature, also I see the issue occurs sporadically, for example it didn't occur in the checks for the PR in the `intel/llvm` repository. If it occurred, I would fill a bug report, no problem.

bader commented:
It's already occurred, so please, file a bug regardless of the status of new run.

ghost commented:
@bader I've opened the issue. I see the checks fail since they don't see latest changes in the sycl runtime/headers. I'm going to re-trigger the checks tomorrow.

bader commented:
Okay. Follow-up on this with @vladimirlaz.

Commit:
[SYCL] Trigger precommit tests

AGindinson reviewed (approved):
FYI, the precommit tests can be retriggered via the Jenkins UI - unless, of course, the re-run requires a later version of test sources to be rebased upon.

ghost commented:
@AGindinson thank you for the information.

ghost commented:
The failed tests aren't related to the introduced changes.
